### ISSUE TITLE
[Elixir] Suggest new exercise: bitstrings

### ISSUE THREAD

**Issue by neenjaw** [exercism/v3#1505]
This issue describes how to implement the `bitstrings` concept exercise for the Elixir track.

[🔗 **How to help**](#actionable-items)

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise to learn what bitstrings are and how to work with them in Elixir

## Learning objectives

- Know what a bitstring is
- Know how to specify a bitstring literal
- Know about the `<<>>` special form
- Know about the `::` binary size modifier

## Out of scope

- bitstrings
- strings

## Concepts

- `bitstrings`
- `tail-recursion`

## Prerequisites

- `basics`
- `recursion`

## Narrative for exercise

Following the [DNA-RNA](https://github.com/exercism/v3/blob/master/reference/stories/primitive-types.rna-and-dna-processing.md) story using bitstring to represent the different nucleotides. 

See this [Julia-lang reference](https://github.com/BioJulia/BioSymbols.jl/blob/master/src/nucleicacid.jl) for how this has been implemented in Julia using an 8 bit block for each nucleotides. Given the flexibility of bitstrings, we can use 4 bits to encode the nucleotides. 

## Resources to refer to

- <https://elixir-lang.org/getting-started/binaries-strings-and-char-lists.html#bitstrings>
- <https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%3C%3C%3E%3E/1>

### Hints

- slight hints to guide toward the solution without giving away exact functions/code/implmentation

### After

- recap about bitstrings, specialforms, modifier

## Representer

- TBD

## Analyzer

- TBD

## Actionable items <a href="#actionable-items" id="actionable-items">🔗</a>

looking for help in a few ways, and I am here to help you through getting this exercise off the ground:

- [ ] Find an applicable problem of small scope for bitstrings
- [ ] Wrap the problem in an arbitrary narrative
- [ ] Write the exemplar solution (How can we fulfil the premise of the exercise with following the scope of requirements)
  - If we can't fulfil the premise with the current scope, then we need to adjust:
    - the premise
    - the code
    - the requirements
  - let's talk about it! We can help you!
- [ ] Writing the test cases (if TDD is your thing!)
- [ ] Writing the student facing documents
  - I am happy to do this for/with you!

For a first phase review we need:

- [ ] the [design.md document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-metadesignmd-file) (the information in this post)
- [ ] the exemplar solution
- [ ] the [introduction document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-docsintroductionmd-file) describing the concepts to be taught

Thanks for considering!

## Need Help?

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by neenjaw** [comment]
Admittedly, this one is a bit more open-ended, I am putting it up here so that it isn't missed since I am working on a follow-up exercise which requires this concept.

**Comment by neenjaw** [comment]
I am going to probably use an accumulator in this problem so it may resolve #1359 at the same time.
